sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -755,8 +571,16 @@ def upload_blob(
                 :dedent: 12
                 :caption: Upload a blob to the container.
         """
-        options = self._upload_blob_options(
-            data,
+        if self.require_encryption and not self.key_encryption_key:
+            raise ValueError("Encryption required but no key was provided.")
+        if kwargs.get('cpk') and self.scheme.lower() != 'https':
+            raise ValueError("Customer provided encryption key must be used over HTTPS.")
+        options = _upload_blob_options(
+            encryption_options=self._build_encryption_options(),
```

Comment:
Ok sorry about this but now that this is only used in two places, can we just pass it inline rather than making a class-level helper?

```py
options = _upload_blob_options(
    encryption_options={
        'required': self.require_encryption,
        'version': self.encryption_version,
        'key': self.key_encryption_key,
         'resolver': self.key_resolver_function
    },
    config=self._config,
    ...
```